File: backend/hisar-hospital-api/src/main/java/com/hisarhospital/hisar_hospital_api/entity/Doctor.java
package com.hisarhospital.hisar_hospital_api.entity;

import jakarta.persistence.*;
import lombok.*;


/**
 * @author adilinan
 */

@Entity
@Table(name = "doctor_table")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Doctor {
    @Id
    private Long id;
    private String qualification;
    private String specialization;
    @Lob
    private String bio;
    private String practiceLocation;
    private String photoUrl;
    @OneToOne
    @MapsId
    @JoinColumn(name = "id")
    private UserEntity user;

}

File: frontend/HisarHospitalApp/src/types/doctor.ts
export interface Doctor {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  qualification: string;
  specialization: string;
  bio: string;
  practiceLocation: string;
  photoUrl: string;
}

// src/types/doctor.ts

import { UserResponse } from "./user";

export interface DoctorResponse extends UserResponse {
  qualification: string;
  specialization: string;
  bio?: string; // Optional
  practiceLocation?: string; // Optional
  photoUrl?: string; // Optional
}
